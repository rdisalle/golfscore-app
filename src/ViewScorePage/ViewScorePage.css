.ViewScorePage {
    border: 1px solid #4681A6;
    margin: 20px;
    padding: 10px;
}

@media only screen and (min-device-width: 768px) {
    .ViewScorePage {
        margin-right: 50px;
        margin-left: 50px;
    }
}

@media screen and (min-device-width: 1200px) {
    .ViewScorePage {
        margin-right: 200px;
        margin-left: 200px;
    }
}