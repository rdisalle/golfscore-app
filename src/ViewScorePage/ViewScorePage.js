import React, { Component } from 'react';
import './ViewScorePage.css';
import ScoresPage from '../ScoresPage/ScoresPage'
import GolfContext from '../GolfContext'
import PropTypes from 'prop-types' 

class ViewScorePage extends Component {
    static defaultProps = {
        match: {
          params: {}
        }
      };

    static contextType = GolfContext;

    onDelete = () => {
        this.props.history.push('/all-scores/:courseId')
      }

    findScore = (scores=[], scoreId) =>
    scores.find(score => score.id === JSON.parse(scoreId))

  render() {
    const { scores } = this.context
    const { scoreId } = this.props.match.params
    const score = this.findScore(scores, scoreId) || { content: '' }

    return (
        <section className="ViewScorePage">
            <h2 className="CourseName">
                {score.course}
            </h2>
            <ScoresPage
                id={score.id}
                name={score.name}
                score_hole_one={score.score_hole_one}
                score_hole_two={score.score_hole_two}
                score_hole_three={score.score_hole_three}
                score_hole_four={score.score_hole_four}
                score_hole_five={score.score_hole_five}
                score_hole_six={score.score_hole_six}
                score_hole_seven={score.score_hole_seven}
                score_hole_eight={score.score_hole_eight}
                score_hole_nine={score.score_hole_nine}
                score_hole_ten={score.score_hole_ten}
                score_hole_eleven={score.score_hole_eleven}
                score_hole_twelve={score.score_hole_twelve}
                score_hole_thirteen={score.score_hole_thirteen}
                score_hole_fourteen={score.score_hole_fourteen}
                score_hole_fifteen={score.score_hole_fifteen}
                score_hole_sixteen={score.score_hole_sixteen}
                score_hole_seventeen={score.score_hole_seventeen}
                score_hole_eighteen={score.score_hole_eighteen}
                total_score={score.total_score}
                to_par={score.to_par} 
                delete= {this.onDelete}
            />
        </section>
    );
  }
}

ViewScorePage.propTypes = {
    match: PropTypes.object
  };

export default ViewScorePage;