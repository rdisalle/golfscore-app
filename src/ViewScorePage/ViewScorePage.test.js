import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ViewScorePage from './ViewScorePage'

describe(`ViewScorePage component`, () => {
  const props = {
    score: {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Golf Course Score One",
        "course": "Golf Course",
        "score_hole_one": "3", 
        "score_hole_two": "4", 
        "score_hole_three": "5",
        "score_hole_four": "4",
        "score_hole_five": "5", 
        "score_hole_six": "6", 
        "score_hole_seven": "2",
        "score_hole_eight": "4",
        "score_hole_nine": "4",
        "score_hole_ten": "5",
        "score_hole_eleven": "5", 
        "score_hole_twelve": "5",
        "score_hole_thirteen": "4",
        "score_hole_fourteen": "4", 
        "score_hole_fifteen": "4", 
        "score_hole_sixteen": "4",
        "score_hole_seventeen": "4", 
        "score_hole_eighteen": "4",
        "total_score": "80",
        "to_par": "+8"
        },
  }

  it('renders a .ViewScorePage by default', () => {
    const wrapper = shallow(<ViewScorePage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Score with score prop', () => {
    const score = shallow(<ViewScorePage {...props} />)
      .find('Score')
    expect(toJson(score)).toMatchSnapshot()
  })
})