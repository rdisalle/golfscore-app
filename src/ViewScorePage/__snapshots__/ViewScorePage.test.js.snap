// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewScorePage component renders a .ViewScorePage by default 1`] = `
<section
  className="ViewScorePage"
>
  <ScoresPage
    delete={[Function]}
    name=""
    score_hole_eight=""
    score_hole_eighteen=""
    score_hole_eleven=""
    score_hole_fifteen=""
    score_hole_five=""
    score_hole_four=""
    score_hole_fourteen=""
    score_hole_nine=""
    score_hole_one=""
    score_hole_seven=""
    score_hole_seventeen=""
    score_hole_six=""
    score_hole_sixteen=""
    score_hole_ten=""
    score_hole_thirteen=""
    score_hole_three=""
    score_hole_twelve=""
    score_hole_two=""
    to_par=""
    total_score=""
  />
</section>
`;

exports[`ViewScorePage component renders a Score with score prop 1`] = `null`;
